syntax = "proto3";

package cn.seisys.dds;

// 所有消息通用监控topic结构，随着主topic名字变化
message TopicMonInfo {
  string uuid = 1;        // 消息唯一id
  int32 domain = 2;       // 数据所在域
  string srcNodeId = 3;   // 原始来源       
  string srcNodeName = 4; // 回复消息来源节点名
  string toNodeId = 5;    // 一般此发送的管理节点或者独立专用监控节点
  string toNodeName = 6;  // 回复消息来源节点名
  string topicName = 7;
  int64 cycle = 8;        // 统计周期，单位秒，一般为60 或者 300秒
  int64 updateTime = 9;   // 发送消息时间
  int64 nCount = 10;      // 周期内发送总数
  int64 nSize = 11;       // 周期内发送总字节数，byte
  int32 srcType = 12;     // 统计来源，0，未知，1，发送端，2，接收端
  string progarmName=13;     // 应用程序的名字
}